test/test_sparse_csr.py

```diff
@@ -1505,13 +1505,23 @@ def _test(t, x, y):
                 y = torch.randn(dk, dj, dtype=dtype, device=device)
                 _test(t, x, y)
 
```

Comment:
Most of the changes on this PR deal with re organizing the layout checks so they are grouped into one place. In addition to that refactoring the PR enables two new layout combinations for `torch.mm`, namely `dense x csc` and `csc x dense` tested here.